### PR TITLE
[RFR][1LP] Cleanup SSH connections after use

### PR DESCRIPTION
**Pull request by rsnyman**
Purpose or Intent
=================

To prevent hanging SSH connections, close SSH connections when we're finished with them.

### CODE REVIEW

**Comment by psav** [comment]
So looking through this, excellent work BTW, my thoughts are this
1) If we are calling out the the appliance to ask it to do some steps via ssh, no context manager should be used. This will make use of the SSH channel which is a) already open and b) will be closed off during shutdown.
2) If we are connecting to a provisioned machine or any other system, we should put all of those calls inside a context manager. the context manager should then have the ssh_close applied to it, since if we are in a context manager, we expect it to be gone at the end of the block anyway.
@mfalesni @jkrocil @RonnyPfannschmidt thoughts?

**Comment by RonnyPfannschmidt** [comment]
i agree with both points of the assessment

**Comment by dajoRH** [comment]
Lint report for commit 73297680201be028b3def8fdf302b3ca8fb1c82f:

`cfme/tests/infrastructure/test_snapshot.py`:
- :bangbang: Line 134:101: **E501** *line too long (107 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/control/test_alerts.py [hunk 1]
```diff
@@ -171,12 +171,12 @@ def _cleanup_vm():
 
 @pytest.yield_fixture(scope="function")
 def ssh(provider, full_template, vm_name):
-    conn = SSHClient(
+    ssh_client = SSHClient(
```
Comment (psav) [inline cfme/tests/control/test_alerts.py]: Can we use a context manager here and yield it?
```python
with SSHClient(......) as ssh_client:
    yield ssh_client
```

File: cfme/tests/optimize/test_bottlenecks.py [hunk 2]
```diff
@@ -50,7 +50,6 @@ def db_restore(temp_appliance_extended_db):
         ssh.get_file("/home/backups/otsuman_db_bottlenecks/REGION{}".format(ver), region_filename)
         guid_filename = "/tmp/GUID_{}".format(fauxfactory.gen_alphanumeric())
         ssh.get_file("/home/backups/otsuman_db_bottlenecks/GUID{}".format(ver), guid_filename)
-    db_storage.close()
```
Comment (psav) [inline cfme/tests/optimize/test_bottlenecks.py]: I like this change, I think as part of it though, we rename it ```db_storage``` to ```db_storage_ssh``` to make it clearer.

**Comment by dajoRH** [comment]
Lint report for commit 8dca2f59b61ec014df7b55d042edf5151c0bfe6b:

`cfme/tests/control/test_alerts.py`:
- :bangbang: Line 177:5: **E125** *continuation line with same indent as next logical line*

`cfme/tests/optimize/test_bottlenecks.py`:
- :red_circle: Line 41:41: **F821** *undefined name 'db_storage_ssh_hostname'*
- :bangbang: Line 46:101: **E501** *line too long (103 > 100 characters)*
- :bangbang: Line 48:101: **E501** *line too long (106 > 100 characters)*
- :bangbang: Line 50:101: **E501** *line too long (105 > 100 characters)*
- :bangbang: Line 52:101: **E501** *line too long (101 > 100 characters)*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4257e826b01341921e996a306062b686323e0472:

`cfme/tests/control/test_alerts.py`:
- :bangbang: Line 177:5: **E125** *continuation line with same indent as next logical line*

`cfme/tests/optimize/test_bottlenecks.py`:
- :heavy_exclamation_mark: Line 49:92: **W291** *trailing whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/configure/test_register_appliance.py [hunk 3]
```diff
@@ -88,29 +88,29 @@ def unset_org_id():
 
 
 def rhsm_unregister():
```
Comment (psav) [inline cfme/tests/configure/test_register_appliance.py]: These should definitely be reworked to receive the appliance object, but I think this is out of scope for the PR

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by rsnyman** [comment]
None of the failures seem directly related to my changes. I've chatted to @psav and decided to move forward with this PR.
